app/models/event.rb

```diff
@@ -242,26 +250,30 @@ def calculate_progress
   end
 
   ##
-  # Returns end of the event
+  # Returns the room in which the event is scheduled
   #
-  def end_time
-    self.start_time + self.event_type.length.minutes
+  def scheduled_room
+    event_schedule.try(:room)
   end
 
   ##
-  # Returns events that are scheduled in the same room and start_time as event
+  # Returns the start time at which this event is scheduled
   #
-  def intersecting_events
-    room.events.where(start_time: start_time).where.not(id: id)
+  def scheduled_start_time
```

Comment:
Because there are many places where we did event.room so doing event.scheduled_room we keep everything the same. And at the beginning the function event_schedule was private as it was only used for the scheduled_room and scheduled_start_timebut now I am using it in more places and it is public. So, I can eliminate scheduled_room and scheduled_start_ time functions and do everywhere event. event_schedule.try(:room). Do I change it then?